actions/actions.py

```diff
@@ -183,6 +192,72 @@ def validate_province(
     ) -> Dict[Text, Optional[Text]]:
         return self.validate_generic("province", dispatcher, value, self.province_data)
 
+    def validate_location(
+        self,
+        value: Text,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> Dict[Text, Optional[Text]]:
+        if not value:
+            dispatcher.utter_message(template="utter_incorrect_selection")
+            return {"location": None}
+
+        if not config.GOOGLE_PLACES_API_KEY:
+            return {
+                "location": value,
+                "latitude": "null",
+                "longitude": "null",
+            }
+
+        querystring = urlencode(
+            {
+                "key": config.GOOGLE_PLACES_API_KEY,
+                "input": value,
+                "language": "en",
+                "inputtype": "textquery",
+                "fields": "formatted_address,geometry",
+            }
+        )
+        response = requests.get(
+            (
+                f"https://maps.googleapis.com"
+                f"/maps/api/place/findplacefromtext/json?{querystring}"
+            )
+        )
+        location = response.json()
+        if location["candidates"]:
+            formatted_address = location["candidates"][0]["formatted_address"]
+            geometry = location["candidates"][0]["geometry"]["location"]
+            latitude = geometry["lat"]
+            longitude = geometry["lng"]
+            return {
+                "location": formatted_address,
+                "latitude": latitude,
+                "longitude": longitude,
+            }
+        else:
+            dispatcher.utter_message(template="utter_incorrect_location")
+            return {"location": None}
+        return {"location": value}
+
+    def validate_location_confirm(
+        self,
+        value: Text,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> Dict[Text, Optional[Text]]:
+        loc_confirm = self.validate_generic(
+            "location_confirm", dispatcher, value, self.yes_no_data
+        )
+        if loc_confirm["location_confirm"] and loc_confirm["location_confirm"] in [
+            "no",
+            1,
+        ]:
```

Comment:
I think we just need to do a `if loc_confirm["location_confirm"] == "no"` here, the `validate_generic` function handles converting the number to `"no"`, and guarantees that `"location_confirm"` is a key in the response.